projects/rero/ng-core/src/lib/service/local-storage.service.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Invenio angular core
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Injectable } from '@angular/core';
+import { CryptoJsService } from './crypto-js.service';
+import { Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LocalStorageService {
+
+  readonly onSet: Subject<any> = new Subject();
```

Comment:
Subjects have to be private and the service have to had getter which return the subject as an observable.

See this link for more info: 
https://stackoverflow.com/a/36989035

Reply:
done